components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java

```diff
@@ -496,17 +498,28 @@ public OAuthApplicationInfo updateGrantTypesAndCallbackURL(String applicationId,
             throw new APIManagementException(msg, ExceptionCodes.OAUTH2_APP_RETRIEVAL_FAILED);
         }
 
-        if (grantTypes == null || grantTypes.isEmpty() || StringUtils.isEmpty(callbackURL)) {
+        if (grantTypes == null || grantTypes.isEmpty()) {
             String msg = "Both Grant Types list and Callback URL can't be null or empty at once.";
             log.error(msg);
             throw new APIManagementException(msg, ExceptionCodes.OAUTH2_APP_RETRIEVAL_FAILED);
         }
+        if (grantTypes.contains(APIMgtConstants.Oauth2Constants.AUTHORIZATION_CODE_GRANT_TYPE) || grantTypes.contains
+                (APIMgtConstants.Oauth2Constants.CODE_GRANT_TYPE)) {
```

Comment:
what's the difference between these 2?

Reply:
This should be implicit and authorization code grants .